0xPolygon/polygon-edge
# Remove WebSocket subscription of which closed connection

# Description

Relevant Issue: https://github.com/0xPolygon/polygon-edge/issues/486

This PR fixes the issue that errors are repeating after WebSocket subscription was closed from client-side without calling unsubscribe method

```bash
2022-04-03T20:58:58.023+0300 [ERROR] polygon.jsonrpc: Unable to write WS message, websocket: close sent
2022-04-03T20:58:58.023+0300 [ERROR] polygon.filter: Unable to process flush, websocket: close sent
2022-04-03T20:59:00.016+0300 [ERROR] polygon.jsonrpc: Unable to write WS message, websocket: close sent
2022-04-03T20:59:00.016+0300 [ERROR] polygon.filter: Unable to process flush, websocket: close sent
```

The error which returned when sending to a closed connection was not checked. This PR removes the filter when FilterManager notices the connection is closed on sending.

# Changes include


# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Checklist


## Testing


### Manual tests

1. Run Polygon Edge nodes under latest `develop` branch
2. Connect to node by WebSocket and call `eth_subscribe` to subscribe events

```json
{
    "jsonrpc": "2.0",
    "method": "eth_subscribe",
    "params": [
        "newHeads"
    ],
    "id": 1
}
```

3. Close WebSocket connection from cient side without `eth_uninstallFilter` or `eth_unsubscribe`

4. Check logs of the nodes the client connected to and the errors like below are outputting forever

```bash
2022-04-03T20:58:58.023+0300 [ERROR] polygon.jsonrpc: Unable to write WS message, websocket: close sent
2022-04-03T20:58:58.023+0300 [ERROR] polygon.filter: Unable to process flush, websocket: close sent
2022-04-03T20:59:00.016+0300 [ERROR] polygon.jsonrpc: Unable to write WS message, websocket: close sent
2022-04-03T20:59:00.016+0300 [ERROR] polygon.filter: Unable to process flush, websocket: close sent
2022-04-03T20:59:02.016+0300 [ERROR] polygon.jsonrpc: Unable to write WS message, websocket: close sent
2022-04-03T20:59:02.016+0300 [ERROR] polygon.filter: Unable to process flush, websocket: close sent
```

5. Checkout the branch of this PR and try to check if such errors no longer appear

# Documentation update

Please link the documentation update PR in this section if it's present, otherwise delete it

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it
